### ISSUE TITLE
Goimports plugin

### ISSUE THREAD

**Issue by mattetti** [constabulary/gb#61]
While I really like the direction gb is taking, the main obstacle to adoption for me is the fact that it breaks my tool chain, more specifically goimports which kicks in every time is save a go file inside vim.

**Comment by davecheney** [comment]
yup, won't be hard, it'll end up looking like
```
gb goimports $IMPORTPATH # applies goimports to $IMPORTPATH
go goimprots # applies goimports to the whole of $PROJECT/src
```
Will that work with your workflow ?

**Comment by mattetti** [comment]
@davecheney so I would have to invoke these 2 commands every time I save and before I call goimports?

**Comment by davecheney** [comment]
@mattetti no, those are just to sample invocations, it'll just be one command.

**Comment by mattetti** [comment]
yes, perfect

**Comment by Xe** [comment]
I've ended up doing things like:

```sh
export GOPATH=$(pwd):$(pwd)/vendor
export PATH=$(pwd)/bin:$PATH
```

and sourcing that when I am in a gb project.

**Comment by Xe** [comment]
Or even better, stick this in your `.lvimrc` for a project:

```vimscript
let $GOPATH = getcwd() . ":" . getcwd() . "/vendor"
```

**Comment by davecheney** [comment]
That won't work if you are in any directory but the project room

**Comment by ches** [comment]
@Xe: something like this might be more robust against Vim's working directory changing, as @davecheney pointed out:

```vim
" Guard multiple execution for currently edited file
if g:localvimrc_sourced_once_for_file | finish | endif

" Guard multiple execution for the running vim instance
if g:localvimrc_sourced_once | finish | endif

" -- Once per Vim instance --
let $GOPATH = g:localvimrc_script_dir . ':' . g:localvimrc_script_dir . '/vendor'
```

See `:help localvimrc-variables`.

vim-go master is now supposed to have some auto-`GOPATH` functionality when detecting a Godep or gb project, but it doesn't seem to work for me yet. See #42 for some links.

**Comment by ches** [comment]
@davecheney: The gb-goimports plugin from #80 is still on a branch—is that pending work on pulling out some of the reusable code in that PR for other plugins to use, as mentioned on the thread, or some other undecided considerations for plugins?

**Comment by davecheney** [comment]
I think that branch is dead, sadly, I don't think it would merge any more.
